src/components/modal/modal.js

```diff
@@ -0,0 +1,39 @@
+import { createPortal } from 'react-dom';
+import PropTypes from 'prop-types';
+import modalStyle from './modal.module.css';
+import ModalOverlay from '../modal-overlay/modal-overlay';
+import { modalRoot } from '../../utils/data';
+import ModalHeader from '../modal-header/modal-header';
+import { useEffect } from 'react';
+
+const Modal = ({children, header, visible, onClose, onKeyDown}) => {
+
+  useEffect(() => {
+    document.addEventListener('keydown', onKeyDown);
+
+    return() => {
+      document.removeEventListener('keydown', onKeyDown);
+    }
+  }, [onKeyDown]);
+  
+  return createPortal (
+    <div className={modalStyle.invisible, visible && modalStyle.visible}>
```

Comment:
Можно лучше:
кажется этот код будет работать не так, как ожидается, `className` ожидает строку, тут можно использовать шаблонный литерал и `modalStyle.visible` добавить в зависимотси от условия:
```suggestion
    <div className={`${modalStyle.invisible} ${visible ? modalStyle.visible : ''}`}>
```